File: backend/Api/Models/Product.cs
using System.Text.Json.Serialization;

namespace Api.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string SKU { get; set; } = string.Empty;
        public string? Brand { get; set; }
        public int? CategoryId { get; set; }
        public decimal Price { get; set; }
        public string? Description { get; set; }
        public bool IsActive { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        public virtual Category? Category { get; set; }

        [JsonIgnore]
        public virtual ICollection<Inventory> InventoryRecords { get; set; } = new List<Inventory>();
        [JsonIgnore]
        public virtual ICollection<SaleItem> SaleItems { get; set; } = new List<SaleItem>();
        [JsonIgnore]
        public virtual ICollection<PurchaseOrderItem> PurchaseOrderItems { get; set; } = new List<PurchaseOrderItem>();
    }
}

File: go-service/models/product.go
package models

import "time"

// Product represents a product in the inventory
type Product struct {
	ID          int        `db:"Id" json:"id"`
	Name        string     `db:"Name" json:"name"`
	SKU         string     `db:"SKU" json:"sku"`
	Brand       *string    `db:"Brand" json:"brand,omitempty"`
	CategoryID  *int       `db:"CategoryId" json:"categoryId,omitempty"`
	Price       float64    `db:"Price" json:"price"`
	Description *string    `db:"Description" json:"description,omitempty"`
	IsActive    bool       `db:"IsActive" json:"isActive"`
	CreatedAt   time.Time  `db:"CreatedAt" json:"createdAt"`
	UpdatedAt   *time.Time `db:"UpdatedAt" json:"updatedAt,omitempty"`
}
